### DISCUSSION TITLE
Guard specific domain pages in domain-driven file strcuture and multiple renderers.

### DISCUSSION THREAD

**Discussion by ashoktamang** [vikejs/vike#1090] [Discussion] [answered]
I am implementing domain-driven file structure with multiple renders. 

The project has two primary domains: `marketing` (SSR mode) and `admin` (SPA mode). In a nutshell, my project structure looks like this:
* domains _(directory name as well)_
  * marketing
    * renderer (SSR mode only)
    * pages
    * _default.page.tsx (route marketing pages to '/')
  * admin
    * renderer (SPA mode only) 
    * pages
      * index.page.route.ts <------ _want to export guard that protects all pages under the route `/admin/*`_
    * _default.page.tsx (route marketing pages to '/admin' inseatd of '/domains/admin') 

## Problem
I can't figure out how to implement page `guard` that protects all the admin pages. If I write individual `index.page.route.ts` files for each route pages, it seems to work. But I would like to implement a single `index.page.route.ts` file that protects all `admin` pages.

Any help or references would be highly appreciated, since project deadline is upon me.

**Comment by brillout** [comment]
Have you tried to define it in `admin/renderer/index.page.route.ts`?

**Reply by ashoktamang** [reply]
I just tried it and its not working. Its not even guarding `/admin` let alone other nested `/admin/*` routes. Basically I am trying to implement an website with an admin panel. The website doesn't need any kind of authentication and all of its pages will be SSR, while all admin pages should be authenticated and will be SPA. How would you implement it?

**Reply by brillout** [reply]
> How would you implement it?

From what you've wrote so far, your approach seems to be right.

> I just tried it and its not working. 

It's a bug then, I'll have a look at it tomorrow. In the meantime, create a guard() hook for each admin page.

**Reply by ashoktamang** [reply]
Thank you for addressing it. Will adhere to your suggestion for the time being.

**Reply by brillout** [reply, marked as answer]
I [fixed a related issue](https://github.com/brillout/vite-plugin-ssr/commit/4013ac66749dc56769596afb468ab9894102bff9) and it should now work (I'll release it later today). Although note that you'll have to use the [V1 design](https://vite-plugin-ssr.com/migration/v1-design) &mdash; more about this later.

**Reply by brillout** [reply]
Released in `0.4.137`. Although, you'll have to migrate to the V1 design to make it work.

> project deadline is upon me

I hope the V1 migration won't take that much time (let me know how that goes). Alternatively, keep using the workround until your deadline, that's probably the fastest.

Btw. would you company be up for [sponsoring](https://github.com/sponsors/brillout)?
